pilota/src/thrift/error/transport.rs

```diff
@@ -0,0 +1,97 @@
+use std::{
+    fmt::{self, Display, Formatter},
+    io,
+};
+
+use faststr::FastStr;
+
+#[deprecated(
+    since = "0.11.0",
+    note = "Please use the `TransportException` instead. This type will be removed in the next release."
+)]
+pub type TransportError = TransportException;
+
+/// Information about I/O errors.
+///
+/// This exception does not send across endpoints, so seems that it is
+/// not necessary to keep it in sync with other languages.
+///
+/// This exception is used to wrap I/O errors and provide a human-readable
+/// error message cache.
+#[derive(Debug)]
+pub struct TransportException {
+    /// Actual I/O error.
+    io_error: io::Error,
+    /// Human-readable error message.
+    message: FastStr,
+}
+
+impl PartialEq for TransportException {
+    fn eq(&self, other: &Self) -> bool {
+        self.io_error.kind() == other.io_error.kind()
+    }
+}
+
+impl Eq for TransportException {}
+
+impl TransportException {
+    #[inline]
+    pub fn io_error(&self) -> &io::Error {
+        &self.io_error
+    }
+
+    #[inline]
+    pub fn message(&self) -> &FastStr {
+        &self.message
+    }
+
+    #[inline]
+    pub fn kind(&self) -> io::ErrorKind {
+        self.io_error.kind()
+    }
+
+    /// Append a message to the existing error message.
+    ///
+    /// That means, the new message will be: `old_message` + `message`.
+    pub fn append_msg(&mut self, message: &str) {
+        let mut s = String::with_capacity(self.message.len() + message.len());
+        s.push_str(self.message.as_str());
+        s.push_str(message);
+        self.message = s.into();
+    }
+
+    /// Prepend a message to the existing error message.
+    ///
+    /// That means, the new message will be: `message` + `old_message`.
+    pub fn prepend_msg(&mut self, message: &str) {
+        let mut s = String::with_capacity(self.message.len() + message.len());
+        s.push_str(message);
+        s.push_str(self.message.as_str());
+        self.message = s.into();
+    }
+}
```

Comment:
These functions are the same for all `Exception`s, maybe we can define a macro for them?